R/models.R

```diff
@@ -93,6 +93,20 @@ predict_model.lda <- function(x, newdata, type, ...) {
     prob = as.data.frame(res$posterior, check.names = FALSE)
   )
 }
+predict_model.H2OModel <- function(x, newdata, type, ...){
+    if (!requireNamespace('h2o', quietly = TRUE)) {
+        stop('The h2o package is required for predicting h2o models')
+    }
+    pred <- h2o.predict(x, h2o::as.h2o(newdata))
+    h2o_model_class <- class(x)[[1]]
+    if (h2o_model_class %in% c("H2OBinomialModel", "H2OMultinomialModel")) {
+        return(as.data.frame(pred[,-1])) 
```

Comment:
Will this return a data.frame with one column per class? (I haven't used h2o myself)

Reply:
Here is some sample output for the multiclass regression. Classes have the class probability, which is consistent with the output of `predict_model` for other classes.
```
> predict_model(model.rf, newdata = df_h2o)
    Iris.setosa Iris.versicolor Iris.virginica
1     1.0000000       0.0000000      0.0000000
2     1.0000000       0.0000000      0.0000000
3     1.0000000       0.0000000      0.0000000
4     1.0000000       0.0000000      0.0000000
5     1.0000000       0.0000000      0.0000000
6     1.0000000       0.0000000      0.0000000
7     1.0000000       0.0000000      0.0000000
8     1.0000000       0.0000000      0.0000000
9     0.8333333       0.1666667      0.0000000
10    1.0000000       0.0000000      0.0000000
```

Reply:
is this based on the standard `iris` dataset? If so, the column names are wrong as they should match the class values (`setosa`, `versicolor`, `virginica`)

Reply:
Yes, that's how h2o does things. I will fix.

Reply:
Thomas, 
I investigated more, and this is not the `iris` dataset that comes with base R. It's a different `iris` dataset that is part of `h2o`, which is why the column names are slightly different than the normal classes. To illustrate:

Here's the output of the `iris` dataset from `h2o`, and you can see the classes are different.
```
> path <- system.file("extdata", "iris.csv", package = "h2o")
> df_h2o <- h2o.importFile(path)
  |==================================================================================================| 100%
> df_h2o
   C1  C2  C3  C4          C5
1 5.1 3.5 1.4 0.2 Iris-setosa
2 4.9 3.0 1.4 0.2 Iris-setosa
3 4.7 3.2 1.3 0.2 Iris-setosa
4 4.6 3.1 1.5 0.2 Iris-setosa
5 5.0 3.6 1.4 0.2 Iris-setosa
6 5.4 3.9 1.7 0.4 Iris-setosa

[150 rows x 5 columns] 
```

Ultimately, the `predict_model` behavior should be acceptable. I ran it on the `iris` dataset that comes with base R. 
```
> lime:::predict_model.H2OModel(model.rf, newdata = iris_h2o)
  |==================================================================================================| 100%
       setosa versicolor   virginica
1   0.9989485 0.00000000 0.001051525
2   0.9989485 0.00000000 0.001051525
3   0.9989485 0.00000000 0.001051525
4   0.9989485 0.00000000 0.001051525
5   0.9989485 0.00000000 0.001051525
6   0.9989485 0.00000000 0.001051525
```